data/graphicsmagick/test.sh

```diff
@@ -0,0 +1,491 @@
+
+
+  # Test 8. Composite images
+
+  # a. Composite an image from two images
+  "gm composite blue.png quadrants500x500_transparent_blue.png __1.png;compare PAE __1.png quadrants500x500.png 0"
+
+  # b. Compute the difference between images:
+  "gm composite -compose difference red.png blue.png __1.png;compare PAE __1.png magenta.png 0"
+
+  # c. Composite an image from two images in a specific position
+  "gm composite -geometry 250x250+250+0 red.png  blue.png -resize 500x250! __1.png; compare PAE __1.png quadrants_up_500x250.png 0"
+
+  # d. Tile a logo across your image:
+  "gm composite -tile logo-primary.png noise.gif __1.png; compare PAE __1.png noise_with_tile_logo.png 0"
+
+
+  # Test 9. Draw shapes and text on a image
+
+  # a. Draw a rectangle in a image
+  "special 0 __1.png"
+
+  #b. Draw a circle in a image
+  "special 1 __1.png"
+
+  # c. Draw text in a image
+  "special 2 __1.png"
+
+  # d. Draw text with variables
+  "special 3 __1.png"
+
+  # e. Draw using a texture
+  "special 4"
+
+
+  # Test 10. Big images
+
+  # a. Scale a image to huge image
+  "resize_and_check degradation.png __1.png - 10000 10000"
+
+  # b. Downgrade a huge image
+  "resize_and_check _-_1.png __1.png - 500 500"
+
+
+  # Test 11.  Batch operations
+  # a. batch http://www.graphicsmagick.org/batch.html
+  "special 5"
+
+  # Test 12. Mogrify
+  # b. bundle operations http://www.graphicsmagick.org/mogrify.html
+  "special 6"
+
+  # Test 13. Test scripting
+  # http://www.graphicsmagick.org/conjure.html
+  "gm conjure -dimensions 10x10 script.html; perl check_size.pl script_test.png 10 10"
+
+)
+
+function special(){
+  case $1 in
+    0) gm convert -resize 500x250! -fill red -draw 'rectangle 250,0 500,250' blue.png $2 && compare PAE $2 quadrants_up_500x250.png 0;;
+    1) gm convert -fill red -draw 'circle 125,125 125,0' white.png $2 && compare PAE $2 red_circle.png 0;;
+    2) gm convert -fill red -draw 'text 0,20 "TEST"' black.png $2 && compare MAE $2 test_text.png 0.01;;
+    3) gm convert -draw 'text 100,100 "%f %wx%h"' white.png $2 && compare MAE $2 test_text_variables.png 0.01;;
+    4) gm convert -size 500x500 xc:red -tile noise.gif -draw 'rectangle 10,10 100,50' __1.png && compare PAE __1.png test_texture.png 0;;
+    5) for file in frame*.png; do
+        outfile=`basename $file .png`.jpg
+        echo convert "'$file'" -rotate 90 \
+        +profile "'*'" "'$outfile'"
+      done | gm batch - && echo -n "OK"
+      ;;
+    6) gm mogrify -format tiff frame*.gif && [ -f "frame1.tiff" ] && \
+       [ -f "frame2.tiff" ] && [ -f "frame3.tiff" ] && [ -f "frame4.tiff" ] &&\
+       echo "OK"
+      ;;
+  esac
+}
+
+######################################
+# Main program
+######################################
+
+count_ok=0
+ok_list=()
+count_ko=0
+ko_list=()
+
+for index in ${!tests[*]}
+do
+    command=${tests[$index]}
+
+    command=$(echo $command | perl -pe "s/(_-_)/.\/$((index-1))-/g")
+    command=$(echo $command | perl -pe "s/(__)/.\/$index-/g")
```

Comment:
like above, this should work:
```sh
command="${command//_-_/.\/$(((index-1)))-}"
command="${command//__/.\/$index-}"
```
also the lines after this could be changed like this

Reply:
Done, thanks